### PR TITLE
Add support for custom fields

### LINKED ISSUE #166: [BUG] Can't read `extra` tags from .m4a files
**Describe the bug**
I added a custom field (via mp3tag) to audio files of various types, but that custom field only appears in tinytag's `extra` collection for the mp3 files and not the m4a files. 

**To Reproduce**
1. Unzip the Songs.zip attachment to find two files - both have a custom tag called `KPL_DATE` with the value `TEST_DATE`
2. Dump the tags for the two contained files using tinytag: `python3 -m tinytag --format json Symphony1.m4a Symphony2.mp3`
3. See that the custom tag is listed in `extra` for the mp3 file but not for the m4a file.  
image

**Expected behavior**
The custom tag is listed under `extra` for both files

**Sample File**
[Songs.zip](https://github.com/devsnd/tinytag/files/10137371/Songs.zip)

### PR DESCRIPTION
**Pull request by mathiascode**
Closes #166

### CODE REVIEW

**Comment by mathiascode** [comment]
@aw-was-here You might be interested in this.

Are there fields you still need that aren't provided as custom fields? We could probably add them to the `extra` dict separately.

File: tinytag/tests/test_all.py [hunk 1]
```diff
@@ -137,8 +143,22 @@
          'audio_offset': 194, 'bitrate': 192.0, 'channels': 2, 'duration': 0.052244897959183675,
          'samplerate': 44100, 'title': 'Tony Hawk VS Wayne Gretzky'}),
     ('samples/id3_xxx_lang.mp3',
-        {'extra': {'isrc': 'USVI20400513', 'lyrics': "Don't fret, precious",
-                   'text': 'SCRIPT\x00\ufeffLatn'},
+        {'extra': {'script': '\ufeffLatn', 'originalyear': '\ufeff2004',
```
Comment (mathiascode) [inline tinytag/tests/test_all.py:146]: Need to take care of extra UTF-16 BOMs somehow.

**Comment by aw-was-here** [comment]
> @aw-was-here You might be interested in this.
> 
> Are there fields you still need that aren't provided as custom fields? We could probably add them to the `extra` dict separately.

Awesome! This PR looks like it fills in a lot of the gaps I've been using another library for in a tremendous way.  Thanks!

**Comment by mathiascode** [comment]
@aw-was-here If you have time, could you try the development version of tinytag, and see how well it covers your needs? Some standard ID3 fields are missing and have to be added manually, let me know which ones you need.

**Comment by aw-was-here** [comment]
@mathiascode I actually started playing with it as soon as it was committed. 😆  So far, nothing sticks out as missing, at least my unit tests pass after making the necessary changes to only use tinytag + how the tags are represented.  That said, I haven't had a chance to double check fields with multiple values to see how they are presented.  But so far so good. Thanks!

**Pull request merged**